server/ntb/commands/sync_topics.py

```diff
@@ -0,0 +1,402 @@
+
+    Command Examples:
+    ::
+
+        $ python manage.py ntb:syn_topics
+        $ python manage.py ntb:syn_topics --file /tmp/iptc-media-topcs.json
+        $ python manage.py ntb:syn_topics --format
+        $ python manage.py ntb:syn_topics --output /tmp/vocabularies.json
+        $ python manage.py ntb:syn_topics --format --output /tmp/vocabularies.json
+    """
+
+    option_list = [
+        superdesk.Option(
+            "--file",
+            "-f",
+            dest="filename",
+            required=False,
+            help="Use a local json file to sync topics from",
+        ),
+        superdesk.Option(
+            "--format",
+            "-t",
+            dest="format_json",
+            action="store_true",
+            required=False,
+            help="Standardise the format of the vocabularies.json file"
+        ),
+        superdesk.Option(
+            "--output",
+            "-o",
+            dest="output",
+            required=False,
+            help="Changes the output file where changes are written. Useful for development."
+        )
+    ]
+
+    def run(self, filename=None, output=None, format_json=False):
+        if format_json:
+            self._run_standardise_vocabs(output)
+        else:
+            self._run_synchronise_vocabs(filename, output)
+
+    def _run_standardise_vocabs(self, output=None):
+        """Standardise the JSON format of the vocabularies.json file
+
+        This allows for a clean diff after synchronising the Media Topics
+        as `json.dumps` may differ from human created changes.
+        """
+
+        logger.info("Standardising the format of vocabularies.json file (skipping sync)")
+
+        vocabularies_json = self._load_vocabularies_json()
+        self._format_json(vocabularies_json, output)
+
+    def _run_synchronise_vocabs(self, filename, output=None):
+        """Synchronise the Media Topics vocab from IPTC"""
+
+        logger.info("Starting to synchronise Media Topics")
+        vocabularies_json = self._load_vocabularies_json()
+
+        existing_topics = self._get_existing_topics_from_cv(vocabularies_json)
+        updated_topics = self._get_updated_topics_from_iptc(filename)
+        report = self._generate_report_data(updated_topics, existing_topics)
+        self._generate_report(report, updated_topics)
+        self._update_vocabularies_json(vocabularies_json, updated_topics)
+        self._write_changes_to_vocabularies_json(vocabularies_json, output)
+
+    def _get_vocabularies_json_path(self):
+        return path.join(app.config["ABS_PATH"], "data", "vocabularies.json")
+
+    def _load_vocabularies_json(self):
+        """Loads the MediaTopics json from the local filesystem"""
+
+        try:
+            with open(self._get_vocabularies_json_path(), "rb") as f:
+                return json.load(f)
+        except IOError:
+            logger.exception("Failed to load vocabularies.json file")
```

Comment:
could be helpful to print the Exception message so we know what's wrong while reading the message.

Reply:
@jerome-poisson 
`logger.exception` logs the tracelog and exception details from the exception that was raised (tested this localle)

Reply:
oh right, fine then, thanks.